core/pkg/evaluator/json.go

```diff
@@ -119,15 +119,16 @@ func (je *JSON) SetState(payload sync.DataSync) (map[string]interface{}, bool, e
 	var events map[string]interface{}
 	var reSync bool
 
+	// TODO: handle other types
```

Comment:
Is this TODO still relevant?

Reply:
Yes... Currently, we don't handle metadata in ADD or UPDATE syncs.

However, there's some other issues with these already, and they are not used in any implementation I know of (they are only really applicable to GRPC). I honestly think we should deprecate them. I will improve the comment and see what others think.

Reply:
I updated this comment; I will open a new discussion about it.